astropy/io/ascii/ipac.py

```diff
@@ -70,20 +70,13 @@ def join(self, vals, widths):
 class IpacHeader(fixedwidth.FixedWidthHeader):
     """IPAC table header"""
     splitter_class = IpacHeaderSplitter
-    col_type_map = {'int': core.IntType,
-                    'integer': core.IntType,
-                    'long': core.IntType,
-                    'double': core.FloatType,
-                    'float': core.FloatType,
-                    'real': core.FloatType,
-                    'char': core.StrType,
-                    'date': core.StrType,
-                    'i': core.IntType,
-                    'l': core.IntType,
-                    'd': core.FloatType,
-                    'f': core.FloatType,
-                    'r': core.FloatType,
-                    'c': core.StrType}
+    col_type_list = (('integer', core.IntType),
+                     ('long', core.IntType),
+                     ('double', core.FloatType),
+                     ('float', core.FloatType),
+                     ('real', core.FloatType),
+                     ('char', core.StrType),
+                     ('date', core.StrType))
```

Comment:
I think this needs a comment that a) the order is important and b) that you decided to handle "d" as double which might be de facto accepted, but contradicts the wording in the standard definition (which days something like "as long as it's still unique ").